src/App.tsx

```diff
@@ -62,6 +69,10 @@ function App(): ReactElement {
     (current: ViewerConfig, newProperties: Partial<ViewerConfig>) => ({ ...current, ...newProperties }),
     defaultViewerConfig
   );
+  const [scatterPlotConfig, updateScatterPlotConfig] = useReducer(
+    (current: ScatterPlotConfig, newProperties: Partial<ScatterPlotConfig>) => ({ ...current, ...newProperties }),
+    getDefaultScatterPlotConfig()
+  );
```

Comment:
The `ViewerConfig` (just above the fold here) uses the same pattern. Using `useReducer` allows `scatterPlotConfig` to be called multiple times in the same state update and reflect all of those changes in the next.

Reply:
do you mean to say "updated multiple times in the same state update"?

Reply:
Yes :X